### ISSUE TITLE
Range.getBookmark returns empty bookmark object

### ISSUE THREAD

**Issue by netomin** [timdown/rangy#232]
Following my implementation of your library (using version: 1.3alpha.20140716) into my android app.
I wanted to try the highlighter module instead of using the serializer and classaplier modules individually (even if they are working great together on my app), but I wanted to take advantage of the useful ready to use serialize and deserialize features of the highlighter that include information about the applied classes in each highlight.

I was not having any js error but the highlight action was not working at all with the highlighter.

Finally I've detected the problem in this function of the default type highlighter:

```
var textContentConverter = {
        rangeToCharacterRange: function(range, containerNode) {
            var bookmark = range.getBookmark(containerNode);
            return new CharacterRange(bookmark.start, bookmark.end);
        },
```

The `range.getBookmark` was always giving an empty object with (start:0 and end:0) then the `return new CharacterRange(bookmark.start, bookmark.end);` line was always returning a zero start and end CharacterRange object and etc, etc so the highlight action was done but invisible.

Maybe this is a problem of the android webkit not supporting this `range.getBookmark` function or maybe not implementing it completely well...
I did some googling about this function and found that there is only supported in IE but your html page demo is working with desktop Chrome so I don't know...

Anyway I revert your code to the previous version:

```
   var textContentConverter = {
        rangeToCharacterRange: function(range, containerNode) {
            //var bookmark = range.getBookmark(containerNode);
            //return new CharacterRange(bookmark.start, bookmark.end);
               var preSelectionRange = range.cloneRange();
               preSelectionRange.selectNodeContents(containerNode);
               range = preSelectionRange.intersection(range);
               var start = 0, end = 0;
               if (range) {
                    preSelectionRange.setEnd(range.startContainer, range.startOffset);
                    start = preSelectionRange.toString().length;
                    end = start + range.toString().length;
                    preSelectionRange.detach();
               }
               return new CharacterRange(start, end);
        },
```

And highlighting using the highlighter module is working ok now.

Then a last question is: Do you think I need to revert also the `characterRangeToRange` function that is using `moveToBookmark` function?

```
characterRangeToRange: function(doc, characterRange, containerNode) {
            var range = api.createRange(doc);
            range.moveToBookmark({
                start: characterRange.start,
                end: characterRange.end,
                containerNode: containerNode
            });

            return range;
        },
```

Thanks.
